core/src/main/scala/kafka/controller/ControllerContext.scala

```diff
@@ -126,6 +130,23 @@ class ControllerContext {
     replicaStates.clear()
   }
 
+  def addTopicId(topic: String, id: UUID): Unit = {
+    topicIds.get(topic).foreach { existingId =>
+      if (!existingId.equals(id))
+        throw new IllegalStateException("topic ID map already contained ID for topic "
+          + topic + " and new ID " + id + " did not match existing ID "
+          + existingId)
+    }
+    topicIds.put(topic, id)
```

Comment:
Do we want to do both checks first and update the two maps together at the end? We may also want to check that the topic exists in `allTopics`.

Reply:
Yeah. That makes sense to me.